client/app/components/conference_room/provider/EventProvider.jsx

```diff
@@ -90,6 +91,7 @@ export default class EventProvider extends React.Component {
     return (
       <Component currentEvent={this.state.currentEvent}
                  nextEvents={this.state.nextEvents}
+                 allEvents={this.state.allEvents}
```

Comment:
Why not just pass `nextEvents.concat([currentEvent])`?
Or maybe the other way around - only store `events` in `state` which would be a list of all events and when passing props to `Component` create a function that will extract `current` and `next` events from all events